### PR TITLE
initial version of dkg state breakup

### PR DESCRIPTION
**Pull request by evanmcc**
I need to figure out a better way to test this.

### CODE REVIEW

File: test/dkg_handler.erl [hunk 1]
```diff
@@ -21,7 +21,8 @@
           privkey :: undefined | tpke_privkey:privkey() | tpke_privkey:privkey_serialized(),
           n :: pos_integer(),
           t :: pos_integer(),
-          id :: pos_integer()
+          id :: pos_integer(),
+          bomb = 10 :: integer()
```
Comment (Vagabond) [inline test/dkg_handler.erl]: What is the bomb?

**Comment by Vagabond** [comment]
Can we remove the "serialized" form of the state records, then?

File: src/dkg_hybriddkg.erl [hunk 2]
```diff
@@ -530,112 +530,186 @@ verify_proofs({vss_ready_proofs, Proofs}, DKG = #dkg{n=N, t=T, f=F}) ->
...
+                            shares_acked = SharesAcked,
+                            shares_seen = SharesSeen,
+                            vss_ready_proofs = ReadyProofs,
+                            echo_proofs = Echo_Proofs,
+                            echo_counts = EchoCount,
+                            ready_counts = ReadyCounts,
+                            leader_changing = LeaderChanging,
+                            leader = Leader,
+                            leader_cap = LeaderCap,
                             %% XXX: Only one element is enough?
```
Comment (Vagabond) [inline src/dkg_hybriddkg.erl]: Yes, we can deserialize PBC elements simply with another element on the same curve...

File: test/dkg_handler.erl (hunk 1 above)
Comment (Vagabond) [reply test/dkg_handler.erl]: Still want to know what the bomb is

Comment (evanmcc) [reply test/dkg_handler.erl]: half-baked idea that never went anywhere.  removing.

**Comment by evanmcc** [comment]
we can get rid of all the old serialization path on both PRs, yes.  The restore will fail for a mixed election, but there's no penalty thereafter.

**Comment by evanmcc** [comment]
in the end we need to export the partial serialization map type from relcast, and then change all the types to that.

**Pull request merged**